lib/FrameManager.js

```diff
@@ -53,7 +52,18 @@ class FrameManager extends EventEmitter {
     this._client.on('Runtime.executionContextDestroyed', event => this._onExecutionContextDestroyed(event.executionContextId));
     this._client.on('Runtime.executionContextsCleared', event => this._onExecutionContextsCleared());
     this._client.on('Page.lifecycleEvent', event => this._onLifecycleEvent(event));
+  }
+
+  async initialize() {
+    const [,{frameTree}] = await Promise.all([
+      this._client.send('Page.enable'),
+      this._client.send('Page.getFrameTree'),
+    ]);
     this._handleFrameTree(frameTree);
```

Comment:
We can get frame events in between page Page.getFrameTree and handleFrameTree. What happens to them?

Reply:
> We can get frame events in between page Page.getFrameTree and handleFrameTree. What happens to them?

This is the unfortunate consequence of the current protocol design. The race is already present in the tip of tree; this patch shortens the gap a bit.